Source/WebInspectorUI/UserInterface/Views/FontVariationDetailsSectionRow.js

```diff
@@ -0,0 +1,179 @@
+        this._variationRangeElement.appendChild(this._variationMinValueElement);
+
+        this._variationMaxValueElement = document.createElement("div");
+        this._variationMaxValueElement.className = "variation-maxvalue";
+        this._variationMaxValueElement.textContent = this._formatAxisValueAsString(maximumValue);
+        this._variationRangeElement.appendChild(this._variationMaxValueElement);
+
+        this._inputRangeElement.addEventListener("input", (event) => {
+            this.dispatchEventToListeners(WI.FontVariationDetailsSectionRow.Event.VariationValueChanged, {tag: event.target.name, value: event.target.value});
+        }, {signal: abortSignal});
+
+        this._valueElement = document.createElement("div");
+        this._valueElement.className = "value";
+        this.element.appendChild(this._valueElement);
+
+        this.tag = tag;
+        this.label = name;
+        this.value = value ?? defaultValue;
+
+        this._defaultValue = defaultValue;
+        this._warningMessage = null;
+        this._warningElement = null;
+    }
+
+    // Public
+
+    get tag()
+    {
+        return this._tag;
+    }
+
+    set tag(value)
+    {
+        this._tag = value;
+        this._tagElement.textContent = value;
+    }
+
+    get label()
+    {
+        return this._label;
+    }
+
+    set label(label)
+    {
+        this._label = label || "";
+
+        if (this._label instanceof Node) {
+            this._labelElement.removeChildren();
+            this._labelElement.appendChild(this._label);
+        } else
+            this._labelElement.textContent = this._label;
+    }
+
+    get value()
+    {
+        return this._value;
+    }
+
+    set value(value)
+    {
+        this._value = value ?? this._defaultValue;
+
+        this._inputRangeElement.value = this._value;
+        this._valueElement.textContent = this._formatAxisValueAsString(this._value);
+    }
+
+    get warningMessage()
+    {
+        return this._warningMessage;
+    }
+
+    set warningMessage(message)
+    {
+        if (this._warningMessage === message)
+            return;
+
+        this._warningMessage = message;
+        this.element.title = this._warningMessage ?? "";
+
+        if (!this._warningElement) {
+            this._warningElement = document.createElement("div");
+            this._warningElement.className = "warning";
+        }
+
+        if (!this._warningElement.parentNode && this._warningMessage)
+            this.element.appendChild(this._warningElement);
+        else
+            this._warningElement.remove();
+    }
+
+    // Private
+
+    _formatAxisValueAsString(value)
+    {
+        const options = {
+            minimumFractionDigits: 0,
+            maximumFractionDigits: 2,
+            useGrouping: false,
+        }
+        return value.toLocaleString(undefined, options);
+    }
+
+    _getAxisResolution(min, max)
+    {
+        let delta = parseInt(max, 10) - parseInt(min, 10);
```

Comment:
The `min` and `max` from a `WI.FontVariationAxis` should always be a number, so it shouldn't be necessary to use `parseInt`. If it is necessary I'm very curious what causes it to be so.

Reply:
Ah. If we just want an integer value we should probably use `Math.round()`. Also I'm not sure the floating point rounding error would cause much of an issue here since we are only using the range internally here to figure out how many fractional figures to display for a value.